hawkular-alerts-engine/src/main/resources/org/hawkular/alerts/engine/rules/ConditionMatch.drl

```diff
@@ -260,6 +264,33 @@ rule Rate
         insert( ce );
 end
 
+rule UpdateMissingStateFromData
+    when
+        $md : MissingState( $tenantId : tenantId, $tsource : source, $did : dataId, previousTime < time )
+        $d  : Data( tenantId == $tenantId, source == $tsource, id == $did)
+   then
+        if (log != null && log.isDebugEnabled()) {
+            log.debugf("UpdateMissingStateFromData: %s ", $md);
+        }
+        modify( $md ) {

```

Comment:
Please comment, either here or somewhere in AlertsEngineImpl that this change is reflected in AlertsEngineImpl, it's important to realize this toi understand why checkMissingState works.